devito/operator.py

```diff
@@ -131,10 +129,6 @@ class Operator(Callable):
     def __init__(self, expressions, **kwargs):
         expressions = as_tuple(expressions)
 
-        # Input check
-        if any(not isinstance(i, Eq) for i in expressions):
```

Comment:
we don't want to drop this

Reply:
perhaps replace `Eq` with `Evaluable` ?

Reply:
and if you think this were a good idea, probably also time to move `Evaluable` from [here](https://github.com/devitocodes/devito/blob/master/devito/tools/abc.py#L145) to `devito/types/evaluable.py` or some other place (most likely still somewhere inside `types.py`

note that after landing #1026 , we now have `Eq` defined in `types.py` too, so it sounds like a nice fit

Reply:
Done the first part - having a check there makes sense. 

I will leave the refactoring about moving `Evaluable` to a later PR.

Reply:
See #1046